python/lsst/meas/algorithms/measureApCorr.py

```diff
@@ -29,14 +29,50 @@
 import lsst.pex.config
 from lsst.afw.image import ApCorrMap
 from lsst.afw.math import ChebyshevBoundedField, ChebyshevBoundedFieldConfig
-from lsst.pipe.base import Task, Struct
+from lsst.pipe.base import Task, Struct, AlgorithmError
 from lsst.meas.base.apCorrRegistry import getApCorrNameSet
 
 from .sourceSelector import sourceSelectorRegistry
 
 
-class MeasureApCorrError(RuntimeError):
-    pass
+class MeasureApCorrError(AlgorithmError):
+    """Raised if Aperture Correction fails in a non-recoverable way.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of the kind of aperture correction that failed; typically an
+        instFlux catalog field.
+    nSources : `int`
+        Number of sources available to the fitter at the point of failure.
+    ndof : `int`
+        Number of degrees of freedom required at the point of failure.
+    iteration : `int`, optional
+        Which fit iteration the failure occurred in.
+    """
+    def __init__(self, *, name, nSources, ndof, iteration=None):
+        msg = f"Unable to measure aperture correction for '{name}'"
+        if iteration is not None:
+            msg += f" after {iteration} steps:"
+        else:
+            msg += ":"
+        msg += f" only {nSources} sources, but require at least {ndof}."
+        super().__init__(msg)
+        self.name = name
+        self.nSources = nSources
+        self.ndof = ndof
+        self.iteration = iteration
+
+    @property
+    def metadata(self):
+        metadata = {"name": self.name,
+                    "nSources": self.nSources,
+                    "ndof": self.ndof,
+                    }
+        # NOTE: have to do this because task metadata doesn't allow None.
+        if self.iteration is not None:
```

Comment:
Is the `is not None` part needed here? I feel like flake8 has yelled at me in the past for not just doing, e.g., `if thing` instead of `if thing is not None`

Reply:
You're thinking of `if thing == None`, which is not good (`None` is a singleton). In this case, the check is explicitly "is this None", not any other "falsey" value.